### PR TITLE
[Issue 742] Automate reporting

### LINKED ISSUE #742: [Task]: Analytics - Automate report posting via a scheduled action
### Summary

Set up a GitHub action to calculate sprint burndown and deliverable percent completion on a regular schedule.

### Acceptance criteria

- [x] The following reports run each day:
  - [x] Sprint burndown
  - [x] Percentage complete by deliverable
- [x] Project maintainers can trigger the analytics reports from GitHub actions without installing the code locally

### PR DESCRIPTION
**Pull request by widal001**
## Summary

Sets up the sprint and deliverable reporting to run on a schedule

Fixes #742 

### Time to review: __10 mins__

- Adds a series of `make` commands that streamline calling the analytics CLI:
  - `make issue-data-export` - Exports issue data from HHS/simpler-grants-gov
  - `make sprint-data-export` - Exports project data from the [Sprint Planning GitHub project](https://github.com/orgs/HHS/projects/13)
  - `make gh-data-export` - Exports both issue and sprint data
  - `make sprint-burndown` - Runs the sprint burndown report
  - `make percent-complete` - Runs the percent complete by deliverable report
  - `make sprint-reports` - Runs both sprint burndown and percent complete
  - `make sprint-reports-with-latest-data` - Exports all data and runs all reports
- Updates the `run-analytics.yml` workflow to run the sprint burndown and percent complete reports and post the results to slack every Monday - Friday at 12:30pm ET
- Adds an end-to-end run of the sprint reports to `ci-analytics.yml` which runs everytime code is pushed to a PR that contains changes to the `analytics/` sub-directory
- Updates the README to describe how to manually trigger the reports from GitHub
- Fixes a bug with how percent complete was calculated

### Notes

- You can trigger the workflow manually from [the GitHub Actions tab](https://github.com/HHS/simpler-grants-gov/actions/workflows/run-analytics.yml). **IMPORTANT** make sure you've selected `issue-742-automate-reporting` as the branch to run the workflow from.
- You can see the additional [instructions in the README](https://github.com/HHS/simpler-grants-gov/blob/3f8b7fa3a3732192335349f4bb543f9f348fc719/analytics/README.md)

### TODO

- In order to enable us to export the project data I had to set up my own personal access token -- we should instead create a GitHub app that we can install for the organization so that we're not using a PAT. That TODO is tracked in: #837

Screenshot 2023-12-08 at 2 13 20 PM

### CODE REVIEW

**Review by sarahknoppA6 — APPROVED** [review]
Thank you for the extremely thorough documentation on this! I think everything looks good to me right now!

File: .github/workflows/ci-analytics.yml [hunk 1]
```diff
@@ -15,6 +15,11 @@ jobs:
   lint-test:
```
Comment (lucasmbrown-usds) [inline .github/workflows/ci-analytics.yml:15]: Putting this here but it's really for the overall PR: your PR explanation is extremely thorough and helpful, well done!

File: analytics/Makefile [hunk 2]
```diff
@@ -1,5 +1,13 @@
 POETRY ?= poetry
 GITHUB ?= gh
+ORG ?= HHS
```
Comment (lucasmbrown-usds) [inline analytics/Makefile:3]: nice

File: analytics/README.md [hunk 3]
```diff
@@ -2,7 +2,33 @@
 
 This sub-directory enables users to run analytics on data generated within the Simpler Grants project.
 
-## Getting Started
+## Getting started
```
Comment (lucasmbrown-usds) [inline analytics/README.md:5]: haha thanks for the war on Title Case

File: analytics/README.md [hunk 4]
```diff
@@ -2,7 +2,33 @@
...
+
+> Note: The following guide will focus on interacting with our analytics package through GitHub and Slack. If you'd like to run this package locally skip to the section on [how to install it locally](#installing-the-analytics-package-locally).
+
+### See the daily reports
```
Comment (lucasmbrown-usds) [inline analytics/README.md:9]: Helpful!

File: analytics/README.md [hunk 5]
```diff
@@ -2,7 +2,33 @@
...
+
+> Note: This option is only available to project maintainers with write access to the repo.
+
+If you're a project maintainer and want to run the reports outside of the daily schedule, you can also trigger the report to run manually using the following steps:
+
+1. Go to the [Run analytics package GitHub Action](https://github.com/HHS/simpler-grants-gov/actions/workflows/run-analytics.yml) page.
+2. Select the "Run workflow" dropdown menu, then click "Run workflow".
+3. For more information about triggering GitHub actions, including running a version of this workflow from another branch, checkout [the GitHub documentation](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow).
+
+<img alt="Screenshot triggering a GitHub action manually" src="./static/screenshot-trigger-gh-action.png" width=750>
```
Comment (lucasmbrown-usds) [inline analytics/README.md:29]: I really appreciate the screenshots!

File: analytics/README.md [hunk 6]
```diff
@@ -50,9 +76,42 @@ If you need to be added to the slack workspace or to the list of collaborators f
...
+<img alt="Screenshot of slack app settings page with bot user OAuth token" src="./static/screenshot-slackbot-token.png" width=750>
 
-## Getting started
+## Using the make commands
+
+In most cases, the reports you'd like to run are already available as `make` commands, specified in our [`Makefile`](./Makefile)
+
+### Export data and run reports
+
+If want to run reports with the most recent data from GitHub, the easiest way to do it with the `make sprint-reports-with-latest-data`
```
Comment (lucasmbrown-usds) [inline analytics/README.md:87]: You probably mean something like `the easiest way to do it *is* with the`. Also period at end of sentence.

File: analytics/src/analytics/datasets/deliverable_tasks.py [hunk 7]
```diff
@@ -110,7 +110,8 @@ def _apply_transformations(
         )
         # calculate task status
         df_all["status"] = "open"
-        is_closed = df_all["closed_date"].isna()
+        # tasks are closed if they DO have a closed_date
+        is_closed = ~df_all["closed_date"].isna()  # ~ is negation
```
Comment (lucasmbrown-usds) [inline analytics/src/analytics/datasets/deliverable_tasks.py:114]: nice catch!

File: analytics/tests/datasets/test_deliverable_tasks.py [hunk 8]
```diff
@@ -99,3 +100,31 @@ def test_keep_30k_deliverables_without_tasks(self):
         # validation - check
         assert df.loc[2, "issue_title"] == "Issue 1"
         assert df.loc[3, "issue_title"] is np.nan
+
+    def test_status_is_closed_if_closed_date_is_none(self):
```
Comment (lucasmbrown-usds) [inline analytics/tests/datasets/test_deliverable_tasks.py:104]: Love that we're keeping the tests up to date!!

**Review by lucasmbrown-usds — APPROVED** [review]
Looks excellent! I have no substantive feedback. Thanks for crafting this so carefully.

File: analytics/README.md (hunk 6 above)
Comment (widal001) [reply analytics/README.md]: Thanks for catching this! Updated it in this commit: [fix: typo in analytics/README](https://github.com/HHS/simpler-grants-gov/pull/835/commits/71914b004683b75e44d7c3c98472262ab48d0c27)

**Review by EOKENAVA — APPROVED** [review]

**Pull request merged**